### PR TITLE
Add TimedRefreshCache

### PR DESCRIPTION
**Pull request by Kloenk**
Add a TimedCache wich TTL is refreshed when the entry is retreived

### CODE REVIEW

**Comment by jaemk** [comment]
Nice - Is this something that could be added as an option to the existing `TimedCache` and `TimedSizedCache` instead of making a new copy though?

**Comment by Kloenk** [comment]
I can try. The borrow checker and mutable was not nice to me, but maybe.

**Comment by Kloenk** [comment]
Should it go into the Trait, e.g. `cached_refresh` and `cached_set_refresh`?

**Comment by jaemk** [comment]
I don't think the trait needs to be updated since it's an impl detail of the particular cache type

File: src/stores/timed.rs [hunk 1]
```diff
@@ -52,9 +47,33 @@ impl<K: Hash + Eq, V> TimedCache<K, V> {
             hits: 0,
             misses: 0,
             initial_capacity: Some(size),
+            refresh: false,
+        }
+    }
+
+    /// Creates a new `TimedCache` with a specified lifespan which
+    /// refreshes the ttl when the entry is retreived
+    pub fn with_lifespan_and_refresh(seconds: u64, refresh: bool) -> TimedCache<K, V> {
```
Comment (jaemk) [inline src/stores/timed.rs:56]: since refresh defaults to false, can it be implied that calling `with_lifespan_and_refresh(60)` would "enable" refresh?

Comment (Kloenk) [reply src/stores/timed.rs:56]: Yes, it could. I would prefer not to do It, maybe the value for refresh comes from some other place and is not constant.

**Comment by Kloenk** [comment]
> I don't think the trait needs to be updated since it's an impl detail of the particular cache type

I just saw that lifespan is in there, so I thought I should at least ask :-)

**Comment by jaemk** [comment]
thanks! released in `0.25.0`

**Pull request merged**